### DISCUSSION TITLE
how to update a python package associated with QGIS-python?

### DISCUSSION THREAD

**Discussion by sunt05** [UMEP-dev/UMEP#409] [Python]
1. locate the actual path to the python interpreter used by your QGIS following [this post](https://github.com/UMEP-dev/UMEP/discussions/408).

say the result is `PosixPath('/Applications/QGIS.app/Contents/MacOS/bin/python3')`, then the path to python should be 

```
/Applications/QGIS.app/Contents/MacOS/bin/python3
```

2. launch the Command-line tool of your system (e.g., `terminal` on macOS) and run the following code to update, for example, `numpy`:

```shell

/Applications/QGIS.app/Contents/MacOS/bin/python3 -m pip install --upgrade numpy

```

then the `numpy` used by QGIS-python should be ungraded.

3. To get this take effect, please restart your QGIS.

**Comment by NishaPatel09** [comment]
Thank You It is helpful
